src/components/Categories/Categories.jsx

```diff
@@ -1,21 +1,35 @@
-import React from 'react'
+import React , {useState, useEffect} from 'react'
 import {CardContainer, CategoriesWrapper, Card} from './Categories.styled';
-import mock2 from '../../assets/mocks/en-us/product-categories.json';
+import {useFeaturedCategories} from '../../utils/hooks/useFeaturedCategories';
+import { Link } from 'react-router-dom';
 
 export default function Categories() {
-    const categories = mock2.results;
+    const [categories, setCategories] = useState([]); 
+    const {data, isLoading} = useFeaturedCategories();
+    
+    useEffect(() => {
+        if(data.results) {
+            setCategories(data.results);
+        }
+    }, [data])
```

Comment:
maybe the dependency should be data.results. If any other data property changes that means that the useEffect is executed